src/libsystemd/sd-journal/journal-authenticate.c

```diff
@@ -284,7 +284,7 @@ int journal_file_hmac_put_object(JournalFile *f, ObjectType type, Object *o, uin
         } else if (type > OBJECT_UNUSED && o->object.type != type)
                 return -EBADMSG;
 
-        sym_gcry_md_write(f->hmac, o, offsetof(ObjectHeader, payload));
+        sym_gcry_md_write(f->hmac, o, sizeof(ObjectHeader));
```

Comment:
i really dislike this btw, because it suggests the size of the prefix would matter instead of the offset of the suffix. yes, of course we are careful, and don't risk the alignment issues, but i think conceptually this is a step back, as we emphasize the wrong thing here in the code.